src/Umbraco.Tests.UnitTests/Umbraco.Web.BackOffice/Controllers/MemberControllerUnitTests.cs

```diff
@@ -0,0 +1,439 @@
+                commonMapper,
+                new CommonTreeNodeMapper(Mock.Of<LinkGenerator>()),
+                new MemberTabsAndPropertiesMapper(
+                    mockCultureDictionary.Object,
+                    backOfficeSecurityAccessor,
+                    textService.Object,
+                    memberTypeService,
+                    memberService,
+                    memberGroupService,
+                    mockPasswordConfig.Object,
+                    contentTypeBaseServiceProvider.Object,
+                    propertyEditorCollection),
+                new HttpContextAccessor());
+
+            var map = new MapDefinitionCollection(new List<IMapDefinition>()
+            {
+                new global::Umbraco.Core.Models.Mapping.MemberMapDefinition(),
+                memberMapDefinition,
+                new ContentTypeMapDefinition(
+                    commonMapper,
+                    propertyEditorCollection,
+                    dataTypeService,
+                    new Mock<IFileService>().Object,
+                    new Mock<IContentTypeService>().Object,
+                    new Mock<IMediaTypeService>().Object,
+                    memberTypeService,
+                    new Mock<ILoggerFactory>().Object,
+                    mockShortStringHelper,
+                    new Mock<IOptions<GlobalSettings>>().Object,
+                    new Mock<IHostingEnvironment>().Object)
+            });
+            _mapper = new UmbracoMapper(map);
+
+            return new MemberController(
+                new DefaultCultureDictionary(
+                    new Mock<ILocalizationService>().Object,
+                    new HttpRequestAppCache(() => null)),
+                new LoggerFactory(),
+                mockShortStringHelper,
+                new DefaultEventMessagesFactory(
+                    new Mock<IEventMessagesAccessor>().Object),
+                textService.Object,
+                propertyEditorCollection,
+                _mapper,
+                memberService,
+                memberTypeService,
+                membersUserManager,
+                dataTypeService,
+                backOfficeSecurityAccessor,
+                new ConfigurationEditorJsonSerializer());
+        }
+
+
+        /// <summary>
+        /// Setup all standard member data for test
+        /// </summary>
+        private Member SetupMemberTestData(
+            out MemberSave fakeMemberData,
+            out MemberDisplay memberDisplay,
+            ContentSaveAction contentAction)
+        {
+            // arrange
+            MemberType memberType = MemberTypeBuilder.CreateSimpleMemberType();
+            Member member = MemberBuilder.CreateSimpleMember(memberType, "Test Member", "test@example.com", "123", "test");
+            int memberId = 123;
+            member.Id = memberId;
+
+            //TODO: replace with builder for MemberSave and MemberDisplay
```

Comment:
I think this can be optimised and use the new builder pattern. I can look into that when I'm further along the line.